src/main.rs

```diff
@@ -53,24 +63,45 @@ fn main() -> Result<()> {
     };
 
     if let Some(Commands::Stats { all }) = cli.command {
+        if cli.query.is_some() {
+            // todo implement --query for stats
+            error!("Sorry, '--query' is not implemented for stats yet.");
+        }
         info!("Getting stats");
         stats::write(
             &util::load_relations(
                 cli.in_file,
-                if all {
-                    util::filter_all_relations
-                } else {
-                    util::filter_target_relations
-                },
+                if all { filter::all } else { filter::by_target },
             )?,
             out,
         )?;
     } else {
         info!("Extracting localities");
-        locality::write(
-            &util::load_relations(cli.in_file, util::filter_target_relations)?,
-            out,
-        )?;
+
+        let filter = {
+            let query_filter = cli.query.as_ref().map(|query| filter::by_query(query));
+
+            move |obj: &OsmObj| -> bool {
+                filter::by_target(obj) && query_filter.as_ref().map_or(true, |f| f(obj))
+            }
+        };
+
+        match cli.format.as_deref() {
+            Some("raw") => {
+                let objs = &util::load_relations(cli.in_file, &filter)?;
```

Comment:
Braucht keine ref oder expliziten deref hier,

```suggestion
                let objs = util::load_relations(cli.in_file, filter)?;
```